terraform/eval_validate.go

```diff
@@ -373,13 +373,15 @@ func (n *EvalValidateResource) Eval(ctx EvalContext) (interface{}, error) {
 	for _, traversal := range n.Config.DependsOn {
 		ref, refDiags := addrs.ParseRef(traversal)
 		diags = diags.Append(refDiags)
-		if len(ref.Remaining) != 0 {
-			diags = diags.Append(&hcl.Diagnostic{
-				Severity: hcl.DiagError,
-				Summary:  "Invalid depends_on reference",
-				Detail:   "References in depends_on must be to a whole object (resource, etc), not to an attribute of an object.",
-				Subject:  ref.Remaining.SourceRange().Ptr(),
-			})
+		if ref != nil {
```

Comment:
Ah ha! It is returning an error if the ref is nil. You can replace this nil check with the following, which is :


```suggestion
		if !refDiags.HasErrors() {
```

@pselle's suggestion below is still accurate :)

Reply:
Thanks, I've changed it in my branch.